cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/SourceControlMeta.java

```diff
@@ -29,18 +29,41 @@ public class SourceControlMeta {
   private final String commitId;
   // The last time the application was synced(push/pull) with git.
   private final Instant lastSyncedAt;
+  @Nullable
+  private final Boolean syncStatus;
 
   /**
    * Default constructor for SourceControlMeta.
    *
-   * @param fileHash the git-hash of the config in git after push
-   * @param commitId the commit in git form/to application was pulled/pushed
+   * @param fileHash     the git-hash of the config in git after push
+   * @param commitId     the commit in git form/to application was pulled/pushed
    * @param lastSyncedAt last time the application was pulled/pushed
    */
-  public SourceControlMeta(String fileHash, @Nullable String commitId, @Nullable Instant lastSyncedAt) {
+  public SourceControlMeta(String fileHash, @Nullable String commitId,
+      @Nullable Instant lastSyncedAt) {
+    this.fileHash = fileHash;
```

Comment:
From what I am interepreting, instead of doing this.syncstatus = null, I should pass it in the constructor.
This would require change in existing code like pullAndDeploy, deploy, pushApp, updateSourceControlMetadata functions and corresponding tests. Is that fine?

Reply:
What I meant you can call `SourceControlMeta(String fileHash, @Nullable String commitId, @Nullable Instant lastSyncedAt, @Nullable Boolean syncStatus)` from inside `SourceControlMeta(String fileHash, @Nullable String commitId, @Nullable Instant lastSyncedAt)` rather than setting fields individually.

Reply:
Thanks for the clarification. Done